python/pyspark/logger/logger.py

```diff
@@ -0,0 +1,150 @@
+# -*- encoding: utf-8 -*-
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import io
+import logging
+import json
+from typing import Any, Optional, Union
+
+
+class JSONFormatter(logging.Formatter):
```

Comment:
Do we guarantee consistency with the structured logger in Spark core? Do we have a test that asserts the presence of the core fields?

Reply:
Added one test for verifying if all keys are properly existing from logger. Thanks!